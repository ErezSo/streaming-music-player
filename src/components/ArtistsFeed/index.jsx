import React from "react";
import { ArtistBox } from "../index";

function ArtistsFeed({ artists }) {
  return (
    <div className="container">
      <h2>Top Artists</h2>
      <hr />
      <div className="list-group">
        {artists.map((artist, i) => <ArtistBox key={i} artist={artist} />)}
      </div>
    </div>
  );
}

export default ArtistsFeed;
