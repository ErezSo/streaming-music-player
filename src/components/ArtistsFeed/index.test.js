import React from "react";
import { shallow } from "enzyme";
import ArtistsFeed from "./";

describe("ArtistsFeed", () => {
  let wrapper;
  const artists = [
    {
      id: "43",
      title: "Bestami Turna - Soundpark 14.04.2018",
      thumb:
        "https://images.hearthis.at/1/5/2/_/uploads/283048/image_track/1914537/w200_h200_q70_----1525319281387.jpg",
      user: {
        username: "TDSmix"
      },
      playback_count: "126"
    }
  ];

  beforeEach(() => {
    wrapper = shallow(<ArtistsFeed artists={artists} />);
  });

  it("renders", () => {
    expect(wrapper).toHaveLength(1);
  });

  it("renders correctly", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
