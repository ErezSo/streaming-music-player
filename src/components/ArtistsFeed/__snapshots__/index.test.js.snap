// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistsFeed renders correctly 1`] = `
"<div className=\\"container\\">
  <h2>
    Top Artists
  </h2>
  <hr />
  <div className=\\"list-group\\">
    <ArtistBox artist={{...}} />
  </div>
</div>"
`;
